ggml-vulkan.cpp

```diff
@@ -0,0 +1,1313 @@
+/**
+ * Copyright (c) 2023 Nomic, Inc. All rights reserved.
+ *
+ * This software is licensed under the terms of the Software for Open Models License (SOM),
+ * version 1.0, as detailed in the LICENSE_SOM.txt file. A copy of this license should accompany
+ * this software. Except as expressly granted in the SOM license, all rights are reserved by Nomic, Inc.
+ */
+
+#include "ggml-vulkan.h"
+#include "ggml.h"
+
+// These are generated at build time by cmake custom command
+#include "shaderop_scale.h"
+#include "shaderop_add.h"
+#include "shaderop_addrow.h"
+#include "shaderop_mul.h"
+#include "shaderop_mulrow.h"
+#include "shaderop_silu.h"
+#include "shaderop_relu.h"
+#include "shaderop_gelu.h"
+#include "shaderop_softmax.h"
+#include "shaderop_norm.h"
+#include "shaderop_rmsnorm.h"
+#include "shaderop_diagmask.h"
+#include "shaderop_mul_mat_f16.h"
+#include "shaderop_mul_mat_q4_0.h"
+#include "shaderop_mul_mat_q4_1.h"
+#include "shaderop_getrows_f16.h"
+#include "shaderop_getrows_q4_0.h"
+#include "shaderop_getrows_q4_1.h"
+#include "shaderop_rope.h"
+#include "shaderop_cpy_f16_f16.h"
+#include "shaderop_cpy_f16_f32.h"
+#include "shaderop_cpy_f32_f16.h"
+#include "shaderop_cpy_f32_f32.h"
+
+#include <iostream>
+#include <vector>
+#include <string>
+#include <memory>
+#include <vector>
+#include <utility>
+#include <fstream>
+#include <exception>
+#include <thread>
+#include <mutex>
+#include <atomic>
+#include <cstring>
+#include <immintrin.h>
+#include <kompute/Kompute.hpp>
+
+#ifndef __STDC_IEC_559__
+#warning Your C implementation does not seem to be IEC 559 compliant, which is required for proper Vulkan interop.
```

Comment:
`#warning`s prevent windows build from working as they are not supported on MSVC (I think it does `#pragma warning`)